src/NuGet.Clients/NuGet.VisualStudio.Common/ErrorListTableDataSource.cs

```diff
@@ -175,15 +178,24 @@ public async Task BringToFrontIfSettingsPermitAsync()
 
             await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            IVsShell vsShell = await _asyncServiceProvider.GetServiceAsync<IVsShell, IVsShell>(throwOnFailure: false);
-            int getPropertyReturnCode = vsShell.GetProperty((int)__VSSPROPID.VSSPROPID_ShowTasklistOnBuildEnd, out object propertyShowTaskListOnBuildEnd);
+            if (!_errorsReported)
```

Comment:
1. Could this just check if `_entries.Count > 0` and get rid of `_errorsReported`?
2. Should this check happen before `SwitchToMainThreadAsync()` or even EnsureInitialized?

Reply:
1. Great simplification, thanks Jeff. I realize now that no other entries are added outside of our own NuGet entries, so yes this makes sense!
2. I had originally thought we need to ensure we're on the UI thread. Now I'm reconsidering and will test to that...
Also thought we needed ErrorList initialized, but this would be a great optimization for scenarios where we never needed to load the error list, now Output logging will not cause the error list to be loaded just to show it unnecessarily!

Reply:
Let's avoid the UI thread switch if we can!

No one will ever complain about that :D